draft_python/mtg_draft_ai/brains.py

```diff
@@ -1,20 +1,163 @@
 """Implementations of Picker, which (hopefully) use intelligent strategies to make draft picks."""
 
+import collections
 import random
+import networkx as nx
+from mtg_draft_ai import synergy
 from mtg_draft_ai.api import Picker
 
 
+COLOR_PAIRS = ['WU', 'WB', 'WR', 'WG', 'UB', 'UR', 'UG', 'BR', 'BG', 'RG']
+COLOR_TRIOS = ['WUB', 'UBR', 'BRG', 'RGW', 'GWU', 'WBR', 'URG', 'BGW', 'RWU', 'GUB']
+
+
 class RandomPicker(Picker):
     """Makes totally random picks. Used for prototyping purposes."""
 
     def pick(self, pack, cards_owned):
         """Makes a random pick from a pack.
 
         Args:
-            pack (list of Card): The current pack to pick a card out of.
-            cards_owned (list of Card): The cards already owned.
+            pack (List[Card]): The current pack to pick a card out of.
+            cards_owned (List[Card]): The cards already owned.
 
         Returns:
             Card: The picked card.
         """
         return pack[random.randint(0, len(pack) - 1)]
+
+
+class Factory:
+
+    def __init__(self, output_class, kwargs):
+        self.output_class = output_class
+        self.kwargs = kwargs
+
+    def create(self):
+        return self.output_class(**self.kwargs)
+
+
+_GSPRating = collections.namedtuple('Rating', ['card', 'color_combo', 'total_edges',
+                                               'new_edges', 'common_neighbors', 'default'])
+
+
+class GreedySynergyPicker(Picker):
+    """Attempts to maximize number of synergy edges, for the color pair with the most edges in its pool."""
```

Comment:
I think that I agree with you in that this is probably not optimal in the long-term, but it's probably fine for now.